lib/matplotlib/colors.py

```diff
@@ -1236,6 +1236,9 @@ def __init__(self, vmin=None, vmax=None, clip=False):
             under colors in a colormap, so it is likely to lead to
             surprises; therefore the default is ``clip=False``.
 
+            *clip* does not have any effect if the range limits *vmin* and *vmax* are
+            not set.
+
```

Comment:
I just realized this is not true. According to the docs, it still has an effect on masked values: "masked values are set to 1"

However, I cannot reproduce the behavior.

![image](https://github.com/matplotlib/matplotlib/assets/2836374/1d30b087-433f-4a7b-a915-617baab4a4e5)

We should first clarify the effect of clip on masked values. If there is really none (which I would see as the correct behavior) and it has been like that for a decent amount of time. We should remove the docs and maksed values.

Then, I'd append the new sentence to the first paragraph because it's in the vmin, vmax context. The following coloring section already moves on in a different direction.